src/coordination/azure/Akka.Coordination.Azure/AzureLease.cs

```diff
@@ -127,53 +127,57 @@ public override Task<bool> Acquire(Action<Exception?>? leaseLostCallback)
-                        }
-
-                        return t.Result switch
-                        {
-                            LeaseActor.LeaseAcquired => true,
-                            LeaseActor.LeaseTaken => false,
-                            Status.Failure f => throw new LeaseException($"Failure while acquiring lease: {f.Cause.Message}", f.Cause),
-                            _ => throw new LeaseException($"Unexpected response type: {t.Result.GetType()}")
-                        };
-                    });
+                    _log.Debug("Acquiring lease [{0}] for [{1}]", _leaseName, _settings.OwnerName);
+                _acquireTask = AcquireTask();
                 
                 return _acquireTask;
             }
+            
+            async Task<bool> AcquireTask()
+            {
+                try
+                {
+                    using var cts = new CancellationTokenSource(_timeout);
+                    var result = await _leaseActor.Ask(new LeaseActor.Acquire(leaseLostCallback), cts.Token);
+                    cts.Token.ThrowIfCancellationRequested();
+
+                    if (result is LeaseActor.LeaseTaken)
+                        _log.Error("Lease {0} for {1} already taken", _leaseName, _settings.OwnerName);
+
+                    return result switch
+                    {
+                        LeaseActor.LeaseAcquired => true,
+                        LeaseActor.LeaseTaken => false,
+                        Status.Failure f => throw new LeaseException($"Failure while acquiring lease [{_leaseName}] for [{_settings.OwnerName}]: {f.Cause.Message}", f.Cause),
+                        _ => throw new LeaseException($"Unexpected response type while acquiring lease [{_leaseName}] for [{_settings.OwnerName}]: [{result.GetType()}] {result}")
+                    };
+                }
+                catch (AskTimeoutException ex)
+                {
+                    throw new LeaseTimeoutException($"Timed out trying to acquire lease [{_leaseName}] for [{_settings.OwnerName}]. It may still be taken.", ex);
+                }
+                catch (AggregateException ex)
+                {
+                    var flattened = ex.Flatten();
+                    if (flattened.InnerExceptions.Any(e => e is AskTimeoutException))
+                        throw new LeaseTimeoutException($"Timed out trying to acquire lease [{_leaseName}] for [{_settings.OwnerName}]. It may still be taken.", ex);
+
+                    throw new LeaseException($"Faulted trying to acquire lease [{_leaseName}] for [{_settings.OwnerName}]. It may still be taken.", ex);
+                }
+                catch (Exception ex)
+                {
+                    throw new LeaseException($"Faulted trying to acquire lease [{_leaseName}] for [{_settings.OwnerName}]. It may still be taken.", ex);
+                }
+                finally
+                {
+                    _acquireTask = null;
```

Comment:
Actual fix, we never reset this cached task, this is the reason why repeated calls to `Acquire()` always returns the same result immediately.

Reply:
LGTM